component/attr_cache/attr_cache.go

```diff
@@ -859,29 +781,24 @@ func (ac *AttrCache) GetAttr(options internal.GetAttrOptions) (*internal.ObjAttr
 	}
 
 	// Get the attributes from next component and cache them
-	pathAttr, err := ac.NextComponent().GetAttr(options)
+	pathAttr, getErr := ac.NextComponent().GetAttr(options)
 
 	ac.cacheLock.Lock()
 	defer ac.cacheLock.Unlock()
 
-	if err == nil {
+	if getErr == nil {
 		// Retrieved attributes so cache them
-		// TODO: bug: when cacheDirs is true, the cache limit will cause some directories to be double-listed
-		// TODO: shouldn't this be an LRU? This sure looks like the opposite...
-		if len(ac.cacheMap) < ac.maxFiles {
-			ac.cacheMap[truncatedPath] = newAttrCacheItem(pathAttr, true, time.Now())
-		} else {
-			log.Debug("AttrCache::GetAttr : %s skipping adding to attribute cache because it is full", options.Name)
-		}
+		ac.cacheMap.insert(pathAttr, true, time.Now())
 		if ac.cacheDirs {
 			ac.markAncestorsInCloud(getParentDir(options.Name), time.Now())
 		}
-	} else if err == syscall.ENOENT {
-		// Path does not exist so cache a no-entry item
-		ac.cacheMap[truncatedPath] = newAttrCacheItem(&internal.ObjAttr{}, false, time.Now())
-	}
+	} else if getErr == syscall.ENOENT {
+		// cache this entity not existing
+		// TODO: change the tests to no longer use empty structs. use internal.createAttr() to define a path instead of a literal.
+		ac.cacheMap.insert(&internal.ObjAttr{Path: internal.TruncateDirName(options.Name)}, false, time.Now())
 
```

Comment:
```suggestion
```
Let's make the code tight by dropping empty lines where it doesn't hurt readability.